install.sh

```diff
@@ -297,8 +311,20 @@ get_release_version() {
 download_hash() {
     HASH_URL=${GITHUB_URL}/download/${VERSION_K3S}/sha256sum-${ARCH}.txt
     info "Downloading hash ${HASH_URL}"
-    curl -o ${TMP_HASH} -sfL ${HASH_URL} || fatal "Hash download failed"
-    HASH_EXPECTED=$(grep " k3s${SUFFIX}$" ${TMP_HASH} | awk '{print $1}')
+    case $DOWNLOADER in
+        curl)
+            curl -o ${TMP_HASH} -sfL ${HASH_URL}
+            ;;
+        wget)
+            wget -qO ${TMP_HASH} ${HASH_URL}
+            ;;
+        *)
+            fatal "Incorrect executable '$DOWNLOADER'"
+            ;;
+    esac
+    # Abort if download command failed
+    [ $? -eq 0 ] || fatal 'Hash download failed'
+    HASH_EXPECTED=`grep " k3s${SUFFIX}$" ${TMP_HASH} | awk '{print $1}'`
```

Comment:
it's better to have a function than having multiple copies.
like:
```
  download() {
      if type curl > /dev/null; then
          curl -o $1 -sfL $2
      else
          # wget equiv     
      fi
  }
```

Reply:
So there you have 2, or actually 3, suggestions in 1:

1) Move the functionality to a separate function instead of having the duplicate copies. I like that.
2a) Do the binary check every time the function is called, instead of once when the script starts. Inefficient and bad way of coding.
2b) Use if/else instead of case: does not scale well if there were to be a 3rd, 4th, etc. fallback option. Already with 2 options the case statement is more readable and it is clear where curl is and where wget is.